### PR TITLE
added thermostat class

### PR DESCRIPTION
**Pull request by matthiaslexer**
Hi there! 

This is my first pull request, so please be kind. :smile: 
At first thanks for your work, you've helped me to finally control my thermostat via an API. 
I've tested this with two of my flourleon thermostats and it works great.

### CODE REVIEW

File: weback_unofficial/thermostat.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
+from weback_unofficial.client import WebackApi, BaseDevice
+
+# uses a weekly plan (Mon, Tues, Wed, Thur, Fri, Sat, Sun)
+AUTO = 'auto'
+# uses the temperature you set in the set_temp property
+MANUAL = 'hand'
+
+WORKING_MODES = {AUTO, MANUAL}
+TEMP_MULTIPLIER = 2
```
Comment (opravdin) [inline weback_unofficial/thermostat.py:9]: You need to describe this constant in comments (what is meant for and why it's 2)

File: weback_unofficial/thermostat.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    def is_available(self):
+        return True if self.shadow.get('connected') == 'false' else True
+
+    @property
+    def mode(self):
+        return self.shadow.get('workmode')
+
+    @property
+    def temperature(self):
+        return self.shadow.get('air_tem') / 10
```
Comment (opravdin) [inline weback_unofficial/thermostat.py:37]: Please, move this 10 value to constants section and add description for it

File: weback_unofficial/thermostat.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+class Thermostat(BaseDevice):
+    def __init__(self, name, client, shadow=None, description=None, nickname=None):
+        super().__init__(name, client, shadow=shadow, description=description, nickname=nickname)
+
+    def setMode(self, mode: str):
+        if not mode in WORKING_MODES:
+            self.raise_invalid_value(WORKING_MODES)
+        
+        print(mode)
```
Comment (opravdin) [inline weback_unofficial/thermostat.py:19]: Remove this print from production version of library please

File: weback_unofficial/thermostat.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            "Mon": self.format_auto_settings('Mon'),
+            "Tue": self.format_auto_settings('Tues'),
+            "Wed": self.format_auto_settings('Wed'),
+            "Thu": self.format_auto_settings('Thur'),
+            "Fri": self.format_auto_settings('Fri'),
+            "Sat": self.format_auto_settings('Sat'),
+            "Sun": self.format_auto_settings('Sun')
+        }
+
+    def format_auto_settings(self, day):
```
Comment (opravdin) [inline weback_unofficial/thermostat.py:59]: It's better to describe input and output of this function, hard to guess from code

File: weback_unofficial/thermostat.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    @property
+    def goal_temperature(self):
+        return self.shadow.get('set_tem') / TEMP_MULTIPLIER
+
+    @property
+    def is_heating(self):
+        return True if self.shadow.get('working_status') == 'on' else False    
+
+    @property
+    def autosettings(self):
```
Comment (opravdin) [inline weback_unofficial/thermostat.py:48]: It's better to move these dow values in constants section, but it's not necessary at this point. I can do it by myself in future versions.

**Review by opravdin — CHANGES_REQUESTED** [review]
I had add some comments in your code. Really appreciate the work you done btw. Also, could you leave me a model of your thermostat? I want to learn more about work modes which you implement here in code.

**Comment by opravdin** [comment]
Cool, thank you for updating your code. I will update this in pypi this week. Could you provide some code examples for me to include in documentation and your thermostat model to include in verified devices list?

**Review by opravdin — APPROVED** [review]
Everything looks fine now, thanks

**Comment by matthiaslexer** [comment]
thanks for the suggestions, fixed all of them. 
It's a Floureon Thermostat (https://www.aliexpress.com/i/32799368790.html). 

If you want to, I can also send you a copy of the shadow and description response.

**Comment by opravdin** [comment]
> If you want to, I can also send you a copy of the shadow and description response.

Yes, that would be perfect! I have plans to add this module to home assistant weback integration, so it will be nice to have this kind of cheatsheet on hands

**Comment by matthiaslexer** [comment]
> > If you want to, I can also send you a copy of the shadow and description response.
> 
> Yes, that would be perfect! I have plans to add this module to home assistant weback integration, so it will be nice to have this kind of cheatsheet on hands

Will send it to you tomorrow. These were also my plans (I only did it for the home assistant integration), so I guess you will get another pull request from me this week.

**Comment by matthiaslexer** [comment]
This is my own documentation, hope you can work with it. 

# Floureon Thermostat Weback Integration

##  Device List
The device list call returns all my thermostats, which are visible in the app.

```json
[
    {
        "Thing_Name": "by-t03-mac-address",
        "Thing_Nick_Name": "Wohnzimmer",
        "Image_Url": "https://app.grit-tech.cn/deviceImage/by/t03/by-t03.png",
        "Sub_type": "by-t03"
    },
    {
        "Thing_Name": "by-t03-mac-address",
        "Thing_Nick_Name": "Gang",
        "Image_Url": "https://app.grit-tech.cn/deviceImage/by/t03/by-t03.png",
        "Sub_type": "by-t03"
    }
]
```
##  Description
The description i've got from aws contains just personal details (e.g. ip adress, location,...) and the following values which seem to be unique for that type of device

```json
{
    "thingTypeName": "_THERMOSTAT",
    "attributes": {
        ...
        "app_name": "WeBack",
        "sub_type": "by-t03",
        ...
    }
}    
```
##  Shadow Object
The shadow object contains all the properties which are also configurable through the app. 
I can describle the following ones: 

**workmode** 
"hand" | "auto"

hand just heats till the set_tem is reached, auto heats accordings to the actions for each day (you can set them up through the app). These actions are saved into Mon, Tues,... (maximum and minimum amount is 6), if you don't set any actions it will just create 6 for you which has the min temperatue (= does nothing)

**tem_uint**
the unit of the temperature (i've got celsius, i guess fahrenheit is also a possible value)

**set_tem** 
the goal temperature of the sensor, it goes from 12 (6 degrees) to 69 (34.5 degrees). Must be a full inte

**air_tem**
the current temperature (also visible on the thermostat), you need to divide it through 10 to get a normal temperature value (211 means 21.1 degrees). 

**safe_lock**
if on you can't change the status of the thermostat on the device itself (a kind of child lock)

**bgl**
controls the background light

**freeze_protect**
automatically turns on heating while below 5 degrees

**controle_mode** 
i've got inter, which menas it uses the internal probe, the app also tells us there is an outer mode (external probe temperature control) and a double mode (two probes simultaneously controlled)

```json
{
    "connected": "true",
    "config_app": "WeBack",
    "firmware_version": "3.2.0rel",
    "offset_hours": 1,
    "offset_minutes": 0,
    "working_status": "on",
    "workmode": "hand",
    "air_tem": 211,
    "floor_tem": 0,
    "set_tem": 70,
    "safe_lock": "on",
    "bgl": "on",
    "poweron_state": "keep",
    "freeze_protect": "off",
    "tem_uint": "C",
    "control_mode": "inter",
    "tem_cal": -20,
    "inter_sen_cal": 10,
    "outer_sen_cal": 30,
    "outer_sen_top": 55,
    "Mon": "04:50_043C,08:00_030C,17:00_043C,20:00_030C,20:00_030C,20:00_030C",
    "Tues": "04:50_043C,08:00_030C,17:00_043C,20:00_030C,20:00_030C,20:00_030C",
    "Wed": "04:50_043C,08:00_030C,17:00_043C,20:00_030C,20:00_030C,20:00_030C",
    "Thur": "04:50_043C,08:00_030C,17:00_043C,20:00_030C,20:00_030C,20:00_030C",
    "Fri": "04:50_043C,08:00_030C,17:00_043C,20:00_030C,20:00_030C,20:00_030C",
    "Sat": "07:20_043C,10:30_030C,17:00_043C,20:00_030C,20:00_030C,20:00_030C",
    "Sun": "07:20_043C,10:30_030C,17:00_043C,20:00_030C,20:00_030C,20:00_030C"
}
```

## Playground Code
```python
from weback_unofficial.client import WebackApi
from weback_unofficial.thermostat import Thermostat, AUTO

client = WebackApi(username, password)
session = client.get_session() # Returns Amazon Session from boto3 package 
devices = client.device_list()
device_name = devices[0]['Thing_Name']

thermostat = Thermostat(device_name, client)
thermostat.update()

# set auto mode (thermoter uses actions which are configured)
thermostat.setMode(AUTO)

# set manual temperature (also sets mode to MANUAL)
thermostat.setTemp(25)

# check if thermostat is currently heating
print(thermostat.is_heating)

# get current temperature
print(thermostat.temperature)
```

**Comment by matthiaslexer** [comment]
@opravdin could you please release a new version on PyPI? Would need it for the home assistant integration.

**Comment by opravdin** [comment]
> @opravdin could you please release a new version on PyPI? Would need it for the home assistant integration.

Yes, sure. I was busy coding the robot vacuum [clean map](https://community.home-assistant.io/t/weback-cloud-integration-testers-and-help-required/186155/67). I will do it today.

**Pull request merged**